### PR TITLE
Refactoring, Adding CLI, Additional Options, etc.

### PR DESCRIPTION
**Pull request by DuckingSimplify**
* Refactored monolithic code into components
* Pulled several occurences of joint fields like album/artist/title into a NamedTuple class for ease/transparency
* Added CLI
  * User can specify output directory -o
  * User can specify to download releases -d [release numbers]
    * without -d : prints the top level info on releases and exits
    * With -d but no numbers: Downloads everything
    * With -d 123 145 : Downloads only releases 123 and 145 - [0 indexed, matching the printed info]
  * User can specify [artist/album/title]-[remove/replace/replace-with] which accept strings of characters and modifies the appropriate field for the output path (not within the metadata/tags)
    *  remove removes all listed characters
    * replace specifies the characters to be replaced
    * replace-with specifies the characters that are being substituted in
    * This is not perfect, but allows a much more configurable experience. For example the starting repo was replacing _ with " " and also dropping a couple of characters.
  * I also added a strip() call in a couple of places, as the starting characters sometimes were spaces.
  
  
`python download_releases.py -o download2 -d 296 314 145 13 --artist-remove "." --artist-replace "øá[]" --artist-replace-with "oa{}" --album-replace " " --album-replace-with "_"`

image
image

### CODE REVIEW

File: download_releases.py [hunk 1]
```diff
@@ -185,3 +206,44 @@
...
+def cli():
+    def is_dir(path):
+        if os.path.isdir(path):
+            return path
+        else:
+            return NotADirectoryError(f"Supplied path is not a directory: {path}")
+
+    parser = argparse.ArgumentParser(prog=__file__, description="Downloader for lofigirl.com")
+    parser.add_argument("-o","--output", help="Output folder", type=is_dir, default="downloads", required=True)
+    parser.add_argument("-d","--download", help="Specify releases to be downloaded. Leave blank for all", type=int, nargs="*")
```
Comment (surfer190) [inline download_releases.py:224]: Running the script with `python download_releases.py -o downloads/`

Lists out all releases and then prints `Skipping download. Specify -d with optional release numbers to download`

It is a common case where a user would want to download all albums the first time the script is run. Can this be accommodated?

**Review by surfer190 — COMMENTED** [review]
Thanks

File: download_releases.py (hunk 1 above)
Comment (DuckingSimplify) [reply download_releases.py:224]: I went to testrun a change for this - turns out I submitted this PR the day before they changed their website. They now have releases posted at https://lofigirl.com/releases/ instead, and there are differences that probably require rethinking/structuring. 

On the bright side, they eliminated the login/registration requirement to download albums straight from the website (and now as ZIP), but it's still 1 by 1 for each album.

**Comment by surfer190** [comment]
Merged the changes in. Unfortunately the site has changed quite a bit lately.
Releases are now at `https://lofigirl.com/releases/` and the old release url 404s.

**Pull request merged**